cache-external-data/action.yml

```diff
@@ -0,0 +1,19 @@
+name: "Cache CMake External Data"
+description: "Caches the external data that we fetch via CMake"
+inputs:
+  store-key-base:
+    required: False
+    description: The base of the store key (will be suffixed with todays date)
+    default: "external-data"
+
+runs:
+  using: "composite"
+  steps:
+  - shell: bash
+    run: echo "NOW=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
```

Comment:
In case of EOS not working this is useless since it doesn't reuse the cache from the previous day. In the `key4hep-build` action this is done on purpose because after changing the nightlies the caches are invalidated since all the paths change but here it could always be the same cache, no?